### PR TITLE
Feature/use context

### PR DESCRIPTION
**Pull request by DrewBradley**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [x] Adds a new feature
- [ ] Improves an existing feature
- [ ] Fixes a bug
- [ ] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
Build AppProvider and wrap the App component in it.

## *Instructions for reviewer:*
*Where should the reviewer start?*
Lines 1-47 of `src/context/AppContext.js`
Lines 10-15 of `src/index.js`
*How should this be manually tested?*
NA
*Any background context you want to provide?*
*What are the relevant tickets?*

## Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have used tests to prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/context/AppContext.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+export const useApp = () => {
+  return useContext(Context)
+}
+
+const AppProvider = ({children}) => {
+  // global useState
+  const [events, setEvents] = useState([])
+  const [contacts, setContacts] = useState([])
+  const [error, setError] = useState('')
```
Comment (richardltyler) [inline src/context/AppContext.js:13]: I really like being able to share this error and loading data with all of the components

**Review by richardltyler — APPROVED** [review]
Looks awesome! Now we just gotta hook it up to the api

File: src/context/AppContext.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+  const [loading, setLoading] = useState('')
+  const [message, setMessage] = useState('')
+
+  // functions
+
+  const getEvents = () => {
```
Comment (kristenmb) [inline src/context/AppContext.js:19]: It is also possible that the api call could happen in/around the Apollo provider and be passed through, but we will definitely look into that!

**Review by kristenmb — APPROVED** [review]
Looks great, we can discuss exactly what we want to store in Context next.

**Review by kevinhartmann23 — APPROVED** [review]
Great setup, now its time to TIEEEE it all together!

**Pull request merged**